### DISCUSSION TITLE
Array importer doesnt consider index

### DISCUSSION THREAD

**Discussion by IngridJSJ** [Spine-project/Spine-Toolbox#1997] [Discussion]
Hello, 
I was trying to import some data as array with the year as index. 
The tool doesn't read the parameter index name. 
It always assigns the index 1, 2, 3, (even if I specify 3,2,1)
<img width="463" alt="arrayimporter" src="https://user-images.githubusercontent.com/47012366/220861348-b02c32c0-9336-4dd8-81a5-02c46b84b7c5.PNG">
<img width="518" alt="array2" src="https://user-images.githubusercontent.com/47012366/220861577-7f9576e6-cc8b-43d1-9b13-ee6c0257604c.PNG">

Is there a way to change the name of the index? 
Thank you !

**Comment by soininen** [comment]
The data is indexed automatically by integers starting from one when Parameter type is "Array". You cannot set your own indices or "reorder" the array. The "Parameter index name" you've set in the Import mapping is actually the *name* of the index:

![image](https://user-images.githubusercontent.com/19147159/220867764-1b60c967-c65f-451a-8696-1541f299eb61.png)

It looks like the "Map" Parameter type would be more suitable for this use case. It allows you to import the year as is. The import settings would look something like this:

![image](https://user-images.githubusercontent.com/19147159/220871075-066e3dd0-e74f-4610-9d7d-2783b5448e51.png)

In the above, I've set "Value" to "Map" and "Parameter index names" to year. The actual years come from the column pointed by "Parameter indexes". This setup gives me Maps that look like this:

![image](https://user-images.githubusercontent.com/19147159/220871737-204938ba-8596-4b39-9ba5-1821b75f0bf4.png)

**Reply by IngridJSJ** [reply]
I just liked the visualization capabilities of arrays. 
Anyways thank you very much! Maps work perfect
